cfbs/analyze.py

```diff
@@ -0,0 +1,521 @@
+        if is_path_component(path, i_comp):
+            return True
+
+    return False
+
+
+DEFAULT_CHECKSUMS_DICT = {"checksums": {}}
+DEFAULT_FILES_DICT = {"files": {}}
+
+
+def checksums_files(
+    files_dir_path,
+    checksums_dict=None,
+    files_dict=None,
+    ignored_path_components=[],
+):
+    if checksums_dict is None:
+        checksums_dict = DEFAULT_CHECKSUMS_DICT
+    if files_dict is None:
+        files_dict = DEFAULT_FILES_DICT
+
+    for root, _, files in os.walk(files_dir_path):
+        for name in files:
+            full_relpath = os.path.join(root, name)
+            tarball_relpath = os.path.relpath(full_relpath, files_dir_path)
+            file_checksum = file_sha256(full_relpath)
+
+            if contains_ignored_components(full_relpath, ignored_path_components):
+                continue
+
+            if not file_checksum in checksums_dict["checksums"]:
+                checksums_dict["checksums"][file_checksum] = set()
+            checksums_dict["checksums"][file_checksum].add(tarball_relpath)
+
+            if not tarball_relpath in files_dict["files"]:
+                files_dict["files"][tarball_relpath] = set()
+            files_dict["files"][tarball_relpath].add(file_checksum)
+
+    return checksums_dict, files_dict
+
+
+def mpf_vcf_dicts():
+    RI_ARCHIVE_URL = (
+        "https://github.com/cfengine/release-information/archive/refs/heads/main.zip"
+    )
+    RI_SHA1_CHECKSUM = "d1bafca809320df8561005a02438b9fa6ea5b995"
+    mpf_vcf_subpath = (
+        "downloads/github.com/cfengine/release-information/archive/refs/heads/"
+        + RI_SHA1_CHECKSUM
+        + "/release-information-main/masterfiles/"
+    )
+    mpf_vcf_path = os.path.join(cfbs_dir(), mpf_vcf_subpath)
+    if not os.path.exists(mpf_vcf_path):
+        fetch_archive(RI_ARCHIVE_URL, RI_SHA1_CHECKSUM, with_index=False)
+    # TODO the release information checksum needs to be updated on each new release
+    # currently, if the checksum is not manually updated, the old, already downloaded files will continue to be used
+    # and if the old files have not already been downloaded, the download from GitHub will fail
+
+    mpf_versions_json_path = os.path.join(mpf_vcf_path, "versions.json")
+    mpf_checkfiles_json_path = os.path.join(mpf_vcf_path, "checksums.json")
+    mpf_files_json_path = os.path.join(mpf_vcf_path, "files.json")
+
+    mpf_versions_dict = get_or_read_json(mpf_versions_json_path)
+    mpf_versions_dict = mpf_versions_dict["versions"]
+
+    mpf_checksums_dict = get_or_read_json(mpf_checkfiles_json_path)
+    mpf_checksums_dict = mpf_checksums_dict["checksums"]
+
+    mpf_files_dict = get_or_read_json(mpf_files_json_path)
+    mpf_files_dict = mpf_files_dict["files"]
+
+    return mpf_versions_dict, mpf_checksums_dict, mpf_files_dict
+
+
+def filepath_sorted(filepaths):
+    """Returns a list of file paths sorted hierarchically.
+
+    Currently not implemented, sorting regularly instead, TODO."""
```

Comment:
Make a ticket and reference it here? How important is sorting it hierarchically?